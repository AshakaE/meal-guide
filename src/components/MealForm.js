import React from 'react';
import { mealType, dishType, cuisineType } from '../assets/options';

const MealForm = () => {
  const [state, setState] = React.useState({
    findMeal: '',
    mealCategory: '',
    dishCategory: '',
    cuisineCategory: '',
  });
  const handleChange = (e) => {
    const values = e.target.value;
    setState({
      ...state,
      [e.target.name]: values,
    });
    // console.log({ values });
  };
  const handleSubmit = () => {
    const filter = {
      ...state,
    };
    if (filter.mealCategory === '') {
      filter.mealCategory = 'breakfast';
    }
    if (filter.dishCategory === '') {
      filter.dishCategory = 'alcohol-cocktail';
    }
    if (filter.cuisineCategory === '') {
      filter.cuisineCategory = 'american';
    }
    console.log(filter);
    return filter;
  };

  return (
    <div>
      <div>
        <input type="text" name="findMeal" onChange={handleChange} />
        <select name="mealCategory" id="mealCategory" onChange={handleChange}>
          {mealType.map((item) => (
            <option key={item} value={item}>
              {item}
            </option>
          ))}
        </select>
        <select name="dishCategory" id="dishCategory" onChange={handleChange}>
          {dishType.map((item) => (
            <option key={item} value={item}>
              {item}
            </option>
          ))}
        </select>
        <select
          name="cuisineCategory"
          id="cuisineCategory"
          onChange={handleChange}
        >
          {cuisineType.map((item) => (
            <option key={item} value={item}>
              {item}
            </option>
          ))}
        </select>
      </div>
      <button type="submit" onClick={handleSubmit}>
        Search Meal
      </button>
    </div>
  );
};

export default MealForm;
