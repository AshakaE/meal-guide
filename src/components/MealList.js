import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import axios from 'axios';
import MealCard from './MealCard';
import url from '../assets/url';

const MealList = (props) => {
  const { meals } = props;
  React.useEffect(() => {
    axios.get(url('', '', '', '')).then((response) => {
      console.log(response.data.hits);
    });
    return () => {};
  });
  return (
    <div>
      <div>
        <div>
          {/* {meals.map((meal) => {
            if (filter !== 'All') {
              if (filter === meal.mealType) {
                return <MealCard key={meal.id} meal={meal} />;
              }
            } else {
              return <MealCard key={meal.id} meal={meal} />;
            }
            return null;
          })} */}
          {meals.map((meal) => (
            <MealCard key={meal.id} meal={meal} />
          ))}
        </div>
      </div>
    </div>
  );
};

const mapStateToProps = (state) => ({
  meals: state.meals,
  filter: state.filter,
});

MealList.propTypes = {
  meals: PropTypes.arrayOf(PropTypes.object).isRequired,
  // filter: PropTypes.string.isRequired,
};

export default connect(mapStateToProps)(MealList);
