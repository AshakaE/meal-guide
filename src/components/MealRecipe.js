import React from 'react';
import { useParams } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { getRecipe } from '../actions/index';
import Loading from './Loading';
import RecipeCard from './RecipeCard';
import vals from '../assets/values';

const MealRecipe = (props) => {
  const { getRecipe, loading, recipe } = props;
  const { id } = useParams();
  const cas = vals(recipe);
  console.log(cas[0]);
  React.useEffect(() => {
    getRecipe(`${id}`);
  }, [getRecipe]);
  if (loading) {
    return <Loading />;
  }
  return (
    <div>
      {Object.values(cas).map((recipe) => (
        <RecipeCard key={recipe.id} recipe={recipe} />
      ))}
    </div>
  );
};

const mapStateToProps = ({ recipeState: { recipe, loading } }) => ({
  loading,
  recipe,
});

MealRecipe.propTypes = {
  loading: PropTypes.bool.isRequired,
  recipe: PropTypes.arrayOf(PropTypes.object).isRequired,
  getRecipe: PropTypes.func.isRequired,
};

export default connect(mapStateToProps, { getRecipe })(MealRecipe);
