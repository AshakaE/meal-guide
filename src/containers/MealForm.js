import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { mealType, dishType, cuisineType } from '../assets/options';
import { getOptions } from '../actions/index';
import styles from '../assets/css/search.module.css';

const MealForm = (props) => {
  const { getOptions } = props;
  const [state, setState] = React.useState({
    findMeal: '',
    mealCategory: '',
    dishCategory: '',
    cuisineCategory: '',
  });
  const handleChange = (e) => {
    const values = e.target.value;
    setState({
      ...state,
      [e.target.name]: values,
    });
  };

  const handleSubmit = () => {
    const filter = {
      ...state,
    };
    if (filter.mealCategory === '') {
      filter.mealCategory = 'breakfast';
    }
    if (filter.dishCategory === '') {
      filter.dishCategory = 'alcohol-cocktail';
    }
    if (filter.cuisineCategory === '') {
      filter.cuisineCategory = 'american';
    }
    if (filter.findMeal === '') {
      filter.findMeal = 'bread';
    }
    getOptions(filter);
  };

  return (
    <div className={styles.house}>
      <div className={styles.container}>
        <div className={styles.header}>
          <h1>The Meal Guide</h1>
        </div>
        <div className={styles.form}>
          <div className={styles.search}>
            <input
              type="text"
              name="findMeal"
              onChange={handleChange}
              className={styles.searchBox}
              placeholder="Search meal"
            />
          </div>
          <div className={styles.input}>
            <select
              name="mealCategory"
              id="mealCategory"
              onChange={handleChange}
              className={styles.singleInput}
            >
              {mealType.map((item) => (
                <option key={item} value={item}>
                  {item}
                </option>
              ))}
            </select>
            <select
              name="dishCategory"
              id="dishCategory"
              onChange={handleChange}
              className={styles.singleInput}
            >
              {dishType.map((item) => (
                <option key={item} value={item}>
                  {item}
                </option>
              ))}
            </select>
            <select
              name="cuisineCategory"
              id="cuisineCategory"
              onChange={handleChange}
              className={styles.singleInput}
            >
              {cuisineType.map((item) => (
                <option key={item} value={item}>
                  {item}
                </option>
              ))}
            </select>
          </div>
          <button type="submit" id="btn" onClick={handleSubmit} className={styles.btn}>
            Search
          </button>
        </div>
      </div>
    </div>
  );
};

const mapStateToProps = ({ filterState: { options } }) => ({
  options,
});

MealForm.propTypes = {
  getOptions: PropTypes.func.isRequired,
};

export default connect(mapStateToProps, { getOptions })(MealForm);
