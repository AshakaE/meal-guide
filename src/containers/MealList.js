import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import MealCard from '../components/MealCard';
import { getMeals } from '../actions/index';
import Loading from '../components/Loading';
import vals from '../assets/values';
import styles from '../assets/css/results.module.css';

const MealList = (props) => {
  const {
    getMeals, loading, meals, options,
  } = props;

  const cas = vals(meals);

  useEffect(() => {
    const {
      findMeal,
      mealCategory,
      dishCategory,
      cuisineCategory,
    } = options;
    getMeals(findMeal, mealCategory, dishCategory, cuisineCategory);
  }, [options]);

  if (loading) {
    return <Loading />;
  }

  return (
    <div>
      <div>
        <div className={styles.listGrid}>
          {Object.values(cas).map((meal) => (
            <MealCard key={meal.id} meal={meal} />
          ))}
        </div>
      </div>
    </div>
  );
};

const mapStateToProps = ({
  mealState: { meals, loading },
  filterState: { options },
}) => ({ loading, meals, options });

MealList.propTypes = {
  loading: PropTypes.bool.isRequired,
  meals: PropTypes.arrayOf(PropTypes.object).isRequired,
  getMeals: PropTypes.func.isRequired,
  options: PropTypes.objectOf(PropTypes.string).isRequired,
};

export default connect(mapStateToProps, { getMeals })(MealList);
