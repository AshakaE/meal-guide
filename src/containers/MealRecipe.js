import React from 'react';
import { useParams } from 'react-router-dom';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { getRecipe } from '../actions/index';
import Loading from '../components/Loading';
import RecipeCard from '../components/RecipeCard';
import vals from '../assets/values';

const MealRecipe = (props) => {
  const { getRecipe, loading, recipe } = props;
  const { id } = useParams();
  const cas = vals(recipe);

  React.useEffect(() => {
    getRecipe(`${id}`);
  }, [getRecipe]);
  if (loading) {
    return <Loading />;
  }
  return (
    <div>
      {Object.values(cas).map((recipe) => (
        <RecipeCard key={recipe.id} recipe={recipe} />
      ))}
    </div>
  );
};

const mapStateToProps = ({ recipeState: { recipe, loading } }) => ({
  loading,
  recipe,
});

MealRecipe.propTypes = {
  loading: PropTypes.bool.isRequired,
  recipe: PropTypes.arrayOf(PropTypes.object).isRequired,
  getRecipe: PropTypes.func.isRequired,
};

export default connect(mapStateToProps, { getRecipe })(MealRecipe);
