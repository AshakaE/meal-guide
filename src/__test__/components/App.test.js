import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter, Switch, Route } from 'react-router-dom';
import { createStore, combineReducers, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';
import { composeWithDevTools } from 'redux-devtools-extension';
import MealRecipe from '../../containers/MealRecipe';
import MealList from '../../containers/MealList';
import MealForm from '../../containers/MealForm';
import mealReducer from '../../reducers/meal';
import recipeReducer from '../../reducers/recipe';
import filterReducer from '../../reducers/filter';

const middleware = [thunk];

const store = createStore(
  combineReducers({
    mealState: mealReducer,
    recipeState: recipeReducer,
    filterState: filterReducer,
  }),
  composeWithDevTools(applyMiddleware(...middleware)),
);

describe('App', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <>
          <BrowserRouter>
            <Switch>
              <Route exact path="/">
                <MealForm store={store} />
                <MealList store={store} />
              </Route>
              <Route path="/recipe/:id">
                <MealRecipe />
              </Route>
            </Switch>
          </BrowserRouter>
        </>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
