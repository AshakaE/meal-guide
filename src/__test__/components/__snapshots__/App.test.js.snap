// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
Array [
  <div
    className="house"
  >
    <div
      className="container"
    >
      <div
        className="header"
      >
        <h1>
          The Meal Guide
        </h1>
      </div>
      <div
        className="form"
      >
        <div
          className="search"
        >
          <input
            className="searchBox"
            name="findMeal"
            onChange={[Function]}
            placeholder="Search meal"
            type="text"
          />
        </div>
        <div
          className="input"
        >
          <select
            className="singleInput"
            id="mealCategory"
            name="mealCategory"
            onChange={[Function]}
          >
            <option
              value="Breakfast"
            >
              Breakfast
            </option>
            <option
              value="Lunch"
            >
              Lunch
            </option>
            <option
              value="Dinner"
            >
              Dinner
            </option>
            <option
              value="Snack"
            >
              Snack
            </option>
            <option
              value="Teatime"
            >
              Teatime
            </option>
          </select>
          <select
            className="singleInput"
            id="dishCategory"
            name="dishCategory"
            onChange={[Function]}
          >
            <option
              value="Alcohol-cocktail"
            >
              Alcohol-cocktail
            </option>
            <option
              value="Biscuits and cookies"
            >
              Biscuits and cookies
            </option>
            <option
              value="Bread"
            >
              Bread
            </option>
            <option
              value="Cereals"
            >
              Cereals
            </option>
            <option
              value="Condiments and Sauces"
            >
              Condiments and Sauces
            </option>
            <option
              value="Drinks"
            >
              Drinks
            </option>
            <option
              value="Desserts"
            >
              Desserts
            </option>
            <option
              value="Egg"
            >
              Egg
            </option>
            <option
              value="Main course"
            >
              Main course
            </option>
            <option
              value="Omlet"
            >
              Omlet
            </option>
            <option
              value="Pancake"
            >
              Pancake
            </option>
            <option
              value="Preps"
            >
              Preps
            </option>
            <option
              value="Preserve"
            >
              Preserve
            </option>
            <option
              value="Salad"
            >
              Salad
            </option>
            <option
              value="Sandwiches"
            >
              Sandwiches
            </option>
            <option
              value="Soup"
            >
              Soup
            </option>
            <option
              value="Starter"
            >
              Starter
            </option>
          </select>
          <select
            className="singleInput"
            id="cuisineCategory"
            name="cuisineCategory"
            onChange={[Function]}
          >
            <option
              value="American"
            >
              American
            </option>
            <option
              value="Asian"
            >
              Asian
            </option>
            <option
              value="British"
            >
              British
            </option>
            <option
              value="Caribbean"
            >
              Caribbean
            </option>
            <option
              value="Central Europe"
            >
              Central Europe
            </option>
            <option
              value="Chinese"
            >
              Chinese
            </option>
            <option
              value="Eastern Europe"
            >
              Eastern Europe
            </option>
            <option
              value="French"
            >
              French
            </option>
            <option
              value="Indian"
            >
              Indian
            </option>
            <option
              value="Italian"
            >
              Italian
            </option>
            <option
              value="Japanese"
            >
              Japanese
            </option>
            <option
              value="Kosher"
            >
              Kosher
            </option>
            <option
              value="Mediterranean"
            >
              Mediterranean
            </option>
            <option
              value="Mexican"
            >
              Mexican
            </option>
            <option
              value="Middle Eastern"
            >
              Middle Eastern
            </option>
            <option
              value="Nordic"
            >
              Nordic
            </option>
            <option
              value="South Amerian"
            >
              South Amerian
            </option>
            <option
              value="South East Asian"
            >
              South East Asian
            </option>
          </select>
        </div>
        <button
          className="btn"
          id="btn"
          onClick={[Function]}
          type="submit"
        >
          Search
        </button>
      </div>
    </div>
  </div>,
  <div>
    <div>
      <div
        className="listGrid"
      />
    </div>
  </div>,
]
`;
